File: Daily Test/Python/Matrix - Replace Asterisks In Column 18.2.2026.py
r, c = map(int, input().split())

mat = [input().split() for _ in range(r)]
ass = []

# Collect positions of '*'
for i in range(c):
    for j in range(r):
        if mat[j][i] == '*':
            ass.append((j, i))

var = 0
cnt = 0

# Traverse from last column to first
for k in range(c - 1, -1, -1):
    for l in range(r):
        if var >= len(ass):
            break

        cnt += 1
        nx, ny = ass[var]

        if mat[l][k] != '*':
            mat[nx][ny] = mat[l][k]
            mat[l][k] = '*'
            var += 1

    if cnt == len(ass):
        break

for row in mat:
    print(*row)

File: Daily Test/C++/Matrix - Replace Asterisks In Column 18.2.2026.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int r, c;
    cin >> r >> c;

    vector<vector<string>> mat(r, vector<string>(c));
    for(int i = 0; i < r; i++)
        for(int j = 0; j < c; j++)
            cin >> mat[i][j];

    vector<pair<int,int>> ass;

    for(int i = 0; i < c; i++)
        for(int j = 0; j < r; j++)
            if(mat[j][i] == "*")
                ass.push_back({j, i});

    int var = 0, cnt = 0;

    for(int k = c-1; k >= 0; k--) {
        for(int l = 0; l < r; l++) {

            if(var >= ass.size()) break;

            cnt++;
            int nx = ass[var].first;
            int ny = ass[var].second;

            if(mat[l][k] != "*") {
                mat[nx][ny] = mat[l][k];
                mat[l][k] = "*";
                var++;
            }
        }
        if(cnt == ass.size()) break;
    }

    for(auto &row : mat) {
        for(auto &val : row)
            cout << val << " ";
        cout << endl;
    }

    return 0;
}